src/app/clusters/push-av-stream-transport-server/push-av-stream-transport-server.cpp

```diff
@@ -0,0 +1,1133 @@
+#include <app/EventLogging.h>
+#include <app/InteractionModelEngine.h>
+#include <app/clusters/push-av-stream-transport-server/push-av-stream-transport-server.h>
+#include <app/reporting/reporting.h>
+#include <app/util/attribute-storage.h>
+#include <app/util/util.h>
+#include <lib/core/CHIPSafeCasts.h>
+#include <lib/support/DefaultStorageKeyAllocator.h>
+#include <protocols/interaction_model/StatusCode.h>
+
+static constexpr uint16_t kMaxConnectionId = 65535; // This is also invalid connectionID
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::PushAvStreamTransport;
+using namespace chip::app::Clusters::PushAvStreamTransport::Structs;
+using namespace chip::app::Clusters::PushAvStreamTransport::Attributes;
+using namespace Protocols::InteractionModel;
+using chip::Protocols::InteractionModel::Status;
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace PushAvStreamTransport {
+
+PushAvStreamTransportServer::PushAvStreamTransportServer(PushAvStreamTransportDelegate & aDelegate, EndpointId aEndpointId,
+                                                         const BitFlags<Feature> aFeatures) :
+    AttributeAccessInterface(MakeOptional(aEndpointId), PushAvStreamTransport::Id),
+    CommandHandlerInterface(MakeOptional(aEndpointId), PushAvStreamTransport::Id), mDelegate(aDelegate), mFeatures(aFeatures),
+    mSupportedFormats{ SupportedFormatStruct{ ContainerFormatEnum::kCmaf, IngestMethodsEnum::kCMAFIngest } }
+{
+    /* set the base class delegates endpointId */
+    mDelegate.SetEndpointId(aEndpointId);
+}
+
+PushAvStreamTransportServer::~PushAvStreamTransportServer()
+{
+    Shutdown();
+}
+
+CHIP_ERROR PushAvStreamTransportServer::Init()
+{
+    LoadPersistentAttributes();
+
+    VerifyOrReturnError(AttributeAccessInterfaceRegistry::Instance().Register(this), CHIP_ERROR_INTERNAL);
+    ReturnErrorOnFailure(CommandHandlerInterfaceRegistry::Instance().RegisterCommandHandler(this));
+    return CHIP_NO_ERROR;
+}
+
+void PushAvStreamTransportServer::Shutdown()
+{ // Unregister command handler and attribute access interfaces
+    CommandHandlerInterfaceRegistry::Instance().UnregisterCommandHandler(this);
+    AttributeAccessInterfaceRegistry::Instance().Unregister(this);
+}
+
+bool PushAvStreamTransportServer::HasFeature(Feature feature) const
+{
+    return mFeatures.Has(feature);
+}
+
+CHIP_ERROR PushAvStreamTransportServer::ReadAndEncodeSupportedFormats(const AttributeValueEncoder::ListEncodeHelper & encoder)
+{
+    for (const auto & supportsFormat : mSupportedFormats)
+    {
+        ReturnErrorOnFailure(encoder.Encode(supportsFormat));
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR PushAvStreamTransportServer::ReadAndEncodeCurrentConnections(const AttributeValueEncoder::ListEncodeHelper & encoder)
+{
+    for (const auto & currentConnections : mCurrentConnections)
+    {
+        ReturnErrorOnFailure(encoder.Encode(currentConnections));
```

Comment:
Again: this is not what the spec says to do.  See https://github.com/project-chip/connectedhomeip/pull/37787#discussion_r2092187419 (which I am resolving, but leaving this comment to track it still being an issue).

You need to filter the list correctly here to implement the spec.